src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIRequestSmelter.java

```diff
@@ -522,16 +558,53 @@ private IAIState retrieveSmeltableFromFurnace()
     }
 
     /**
-     * Very simple action, straightly extract it from the furnace.
+     * Retrieve used fuel from the furnaces. If no position has been set return. Else navigate to the position of the furnace. On arrival execute the extract method of the
+     * specialized worker.
+     *
+     * @return the next state to go to.
+     */
+    private IAIState retrieveUsedFuel()
+    {
+        worker.getCitizenStatusHandler().setLatestStatus(new TranslationTextComponent(COM_MINECOLONIES_COREMOD_STATUS_RETRIEVING));
+
+        if (walkTo == null)
+        {
+            return START_WORKING;
+        }
+
+        if (walkToBlock(walkTo))
+        {
+            return getState();
+        }
+
+        final TileEntity entity = world.getTileEntity(walkTo);
+        if (!(entity instanceof FurnaceTileEntity)
+                || (ItemStackUtils.isEmpty(((FurnaceTileEntity) entity).getStackInSlot(FUEL_SLOT))))
+        {
+            walkTo = null;
+            return START_WORKING;
+        }
+
+        walkTo = null;
+
+        extractFromFurnaceSlot((FurnaceTileEntity) entity, FUEL_SLOT);
+        return START_WORKING;
+    }
+
+    /**
+     * Very simple action, straightly extract from a furnace slot.
      *
      * @param furnace the furnace to retrieve from.
      */
-    private void extractFromFurnace(final FurnaceTileEntity furnace)
+    private void extractFromFurnaceSlot(final FurnaceTileEntity furnace, final int slot)
     {
         InventoryUtils.transferItemStackIntoNextFreeSlotInItemHandler(
-          new InvWrapper(furnace), RESULT_SLOT,
+          new InvWrapper(furnace), slot,
           worker.getInventoryCitizen());
-        worker.getCitizenExperienceHandler().addExperience(BASE_XP_GAIN);
+        if(slot == RESULT_SLOT)
+        {
+            worker.getCitizenExperienceHandler().addExperience(BASE_XP_GAIN);
+        {
```

Comment:
wat

Reply:
I combined the functions for extracting from a furnace slot for both the fuel and result slots, and I changed it so experience would only be rewarded if an item was taken from the result slot as none should be rewarded from the fuel slot.

Reply:
No, my point was that this doesn't even compile, since the wrong brace was used.

Reply:
I have no idea how that snuck by. I even tested it on my computer and it compiled fine. I fixed it in another PR.